estuary/authorization.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from __future__ import unicode_literals
+
+import ssl
+
+from flask import current_app
+from werkzeug.exceptions import InternalServerError
+
+from estuary import log
+
+
+def is_user_authorized(username, employee_type):
+    """
+    Verify the user is authorized to access the application.
+
+    :param str username: the username from the user's token
+    :param str employee_type: the employee type from the user's token
+    :return: a boolean that determines if the user is authorized
+    :rtype: bool
+    """
+    employee_types = current_app.config.get('EMPLOYEE_TYPES', [])
+    if employee_type in employee_types:
+        log.debug('The user %s is an employee', username)
+        return True
+
+    ldap_group_dn = current_app.config.get('LDAP_EXCEPTIONS_GROUP_DN')
+    if ldap_group_dn and username in _get_exception_users():
+        log.debug('The user %s is not considered an employee but is an exception', username)
+        return True
+
+    return False
+
+
+def _get_exception_users():
+    """
+    Get the list of users that are explicitly whitelisted.
+
+    If the LDAP search fails, an empty set is returned.
+
+    :return: a set of usernames
+    :rtype: set
```

Comment:
What is the advantage of using a set here?

Reply:
No advantage other than it let's the caller of the function know that there are no duplicate values.